### PR TITLE
Variables: Variables Type section added

### PR DESCRIPTION
**Pull request by mottaquikarim**
**Optional**
Noticed that variable types are kinda mentioned in the subsequent sections but no actual intro is given in the variables section. Feel free to reject if you feel like this is TMI. (Which is why this is a separate pull request).
- added a new section called Variable Types
- variable types are never explicitly introduced but mentioned several times in the subsequent sections
- this section is meant to be a super lightweight intro to types, just so the student knows that there are differences between strings and numbers

### CODE REVIEW

File: lecture-variables.html [hunk 1]
```diff
@@ -118,12 +132,54 @@
 someNumber = someNumber / x
 someNumber /= x
 </code></pre>
+    <p>
+(Optional) <b>Challenge</b>: what is the value of <code>someNumber</code> at the end of the above code snippet? Walk through each line of code and try to compute the value of <code>someNumber</code> yourself. Then, paste the snippet above into codepen to check your answer.
+    </p>
+
+    <h2>Variable Types</h2>
```
Comment (bensussman) [inline lecture-variables.html]: I'm down with this. Do you think it makes sense to have this at the end? I do have a (admittedly small) bullet about types up on line 81 above...

**Comment by bensussman** [comment]
This seems like a good addition. This can't be merged due to conflicts though. Can you fix the conflicts (and the indentation!) so I can merge this?

My other concern is: do we think this belongs at the end of this section? I mention types a few times in this section, is that all just buildup to this content at the bottom? Or should we move this content higher? Or kill the mentions of types till we reach this content?

**Comment by mottaquikarim** [comment]
Sure thing, I'll get to it most likely Thursday but if I have some downtime tonight I'll knock it out. 

Also, fair points re: placement. I was conservative with where I added this mainly because I wasn't sure if you were holding off on introducing types on purpose. I will merge master into this branch and move placement around, etc. I feel like it should be higher up -- what do you think?

**Comment by bensussman** [comment]
I'm not sure if higher up necessarily makes sense. I'd have to run through it to see what works when being delivered...

**Comment by bensussman** [comment]
Ok i thought about it and i think it belongs above the "good variable names" section. I think we should roll the `good variable names / bad variable names` block down into the "good variable names" section, and precede it with this types section.

I also think we should consider rewriting the "things to know about variables" bullets given that much of their ideas are in subsequent sections in a more fleshed out way. Doesn't mean get rid of it, but maybe rewrite it with that context in mind.

@mottaquikarim can you make this happen?

**Comment by mottaquikarim** [comment]
@bensussman done deal -- made a bunch of changes! I'm pretty happy with how this turned out. 

Main point to note is that the "things to know about variables" bullets **has** been moved out into the subsequent sections that address them. (Except the first point about reusing variables, which now lives in the paragraph above).

I also expanded the Good Variable Names section and the Numbers subsection in Variable Types, hopefully to make some of the points being made a bit easier to read (IMO lists / bullets / example blocks >> walls of text).

Let me know your thoughts!

**Comment by bensussman** [comment]
👍 love it

**Pull request merged**